pkg/util/table.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"github.com/magiconair/properties"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tipb/go-tipb"
+	"math"
+)
+
+type Table struct {
+	columns []tipb.ColumnInfo
+	rowC    *RowCodec
+	tableId int64
+}
+
+func NewTable(p *properties.Properties) *Table {
+	table := &Table{tableId: 1001}
+	table.rowC = NewRowCodec(p)
+	colLen := len(table.rowC.fields)
+	table.columns = make([]tipb.ColumnInfo, 0, colLen)
+	for i := 0; i < colLen; i++ {
+		table.columns = append(table.columns, tipb.ColumnInfo{
+			PkHandle:  false,
+			ColumnId:  int64(i),
+			Collation: int32(mysql.DefaultCollationID),
+			ColumnLen: types.UnspecifiedLength,
+			Decimal:   types.UnspecifiedLength,
+			Tp:        int32(mysql.TypeVarchar)})
+	}
+	return table
+}
+
+func (t *Table) DAGTableScanReq(fields []string) *tipb.DAGRequest {
+	dag := &tipb.DAGRequest{}
+	dag.StartTs = math.MaxInt64
+	output := make([]uint32, 0, len(fields))
+	for i := 0; i < len(fields); i++ {
```

Comment:
For YCSB, fields can be empty which means fetching all the fields.

Reply:
Hava supported the empty fields